lib/html/pipeline/@mention_filter.rb

```diff
@@ -70,7 +70,7 @@ def self.mentioned_logins_in(text, username_pattern=UsernamePattern)
       def call
         result[:mentioned_usernames] ||= []
 
-        doc.search('text()').each do |node|
+        doc.search('.//text() | text()').each do |node|
```

Comment:
Without this change, tests failed with Nokogiri 1.6.6.x:

```
Nokogiri::XML::XPath::SyntaxError:
Invalid expression: .//child::text() | self::child::text()
```

> Please, as a workaround, use #xpath with a real XPath expression (as above) to avoid having to convert CSS into a hacky XPath expression to work around unrelated DocumentFragment issues.
> https://github.com/sparklemotion/nokogiri/issues/1233#issuecomment-71465416

Nokogiri author suggested use real XPath to avoid this.